qiskit/providers/ibmq/utils/utils.py

```diff
@@ -56,3 +59,47 @@ def validate_job_tags(job_tags: Optional[List[str]], exception: Type[Exception])
     if job_tags and (not isinstance(job_tags, list) or
                      not all(isinstance(tag, str) for tag in job_tags)):
         raise exception("job_tags needs to be a list or strings.")
+
+
+def setup_logger(logger: Logger) -> None:
+    """Setup the logger for the provider modules with the appropriate level.
+
+    It involves:
+        * Use the `QISKIT_IBMQ_PROVIDER_LOG_LEVEL` environment variable to
+            determine the log level for the provider modules. If it is set, the
+            provider uses the specified level. If it is set to an invalid level,
+            the log level defaults to `INFO`. The valid log levels are ``DEBUG``,
+            ``INFO``, ``WARNING``, ``ERROR``, and ``CRITICAL`` (case-insensitive).
+        * Use the `QISKIT_IBMQ_PROVIDER_LOG_FILE` environment variable to
+            specify the filename to use for logging the logs to a file. If it is
+            not set, the logs will not be logged to a file.
+    """
+    log_level = os.getenv('QISKIT_IBMQ_PROVIDER_LOG_LEVEL', '')
+    log_file = os.getenv('QISKIT_IBMQ_PROVIDER_LOG_FILE', '')
+
+    # Set the logging level, if specified.
+    if log_level:
+        # Default to `INFO` if the specified level is not valid.
+        level = logging.getLevelName(log_level.upper())
+        if not isinstance(level, int):
+            level = logging.INFO
+        logger.setLevel(level)
+
+    # Although the log level might not be specified, setup the formatter.
+    # This ensures log messages are formatted, when logged to the console, in the
+    # case they propagate to the root logger.
+    log_fmt = ('{}.%(module)s.%(funcName)s:%(levelname)s:%(asctime)s:'
+               ' %(message)s'.format(logger.name))
+    formatter = logging.Formatter(log_fmt)
+
+    # Setup the stream handler, for logging to console, with the given format.
+    stream_handler = logging.StreamHandler()
+    stream_handler.setFormatter(formatter)
+    logger.addHandler(stream_handler)
+
+    # Set the logging file, if specified.
+    if log_file:
+        # Setup the file handler.
+        file_handler = logging.FileHandler(log_file)
+        file_handler.setFormatter(formatter)
+        logger.addHandler(file_handler)
```

Comment:
I'm thinking about not logging to the screen if a log file is specified. This is useful for people who only want to look at the log if there's an error and don't want their screen cluttered.